fapi-client/src/main/scala/com/gu/facia/api/FAPI.scala

```diff
@@ -189,4 +190,87 @@ object FAPI {
     val backfillRequest = BackfillResolver.resolveFromConfig(collectionConfig)
     BackfillResolver.backfill(backfillRequest, adjustSearchQuery, adjustItemQuery)
   }
+
+  private def updateContentProperties(content: FaciaContent, properties: ContentProperties): FaciaContent = content match {
+    case curatedContent: CuratedContent => curatedContent.copy(properties = properties)
+    case supportingCuratedContent: SupportingCuratedContent => supportingCuratedContent.copy(properties = properties)
+    case snap: LatestSnap => snap.copy(properties = properties)
+    case snap: LinkSnap => snap.copy(properties = properties)
+  }
+
+  private def updateGroup(content: FaciaContent, group: String): FaciaContent = content match {
+    case curatedContent: CuratedContent => curatedContent.copy(group = group)
+    case supportingCuratedContent: SupportingCuratedContent => supportingCuratedContent.copy(group = group)
+    case snap: LatestSnap => snap.copy(group = group)
+    case snap: LinkSnap => snap.copy(group = group)
+  }
+
+  private def applyDefaultBoost[T](
+    content: T,
+    default: BoostLevel,
+    allowedBoosts: List[BoostLevel],
+    getBoostLevel: T => BoostLevel,
+    setBoostLevel: (T, BoostLevel) => T
+  ): T =
+    if (allowedBoosts.contains(getBoostLevel(content)))
+      content
+    else
+      setBoostLevel(content, default)
+
+  case class GroupBoostConfig(default: BoostLevel, allowedBoosts: List[BoostLevel], maxItems: Int, group: String)
+
+  private val flexibleGeneralBoosts = List(
+    // Splash
+    GroupBoostConfig(BoostLevel.Default, List(BoostLevel.Default, BoostLevel.Boost, BoostLevel.MegaBoost, BoostLevel.GigaBoost), maxItems = 1, group = "3"),
+    // Very large
+    GroupBoostConfig(BoostLevel.MegaBoost, List(BoostLevel.MegaBoost), maxItems = 0, group = "2"),
+    // Large
+    GroupBoostConfig(BoostLevel.Boost, List(BoostLevel.Boost, BoostLevel.MegaBoost), maxItems = 0, group = "1"),
+    // Standard
+    GroupBoostConfig(BoostLevel.Default, List(BoostLevel.Default, BoostLevel.Boost, BoostLevel.MegaBoost), maxItems = 8, group = "0")
+  )
+
+  private def boostsConfigFor(collectionType: String, groupsConfig: List[GroupConfig]): Option[List[GroupBoostConfig]] = {
+    val groupsConfigBiggestFirst = groupsConfig.reverse
```

Comment:
`BiggestFirst` feels ambiguous as aside from the size of the item (e.g. splash) it made me think of the number of items in the group... Maybe something alluding to the prominence of the stories like BiggestDisplayFirst?

Reply:
`groupsConfigTopGroupFirst`?